tracer/src/Datadog.Trace/Sampling/SpanSamplingRule.cs

```diff
@@ -0,0 +1,157 @@
+
+            _serviceNameRegex = ConvertGlobToRegex(serviceNameGlob);
+            _operationNameRegex = ConvertGlobToRegex(operationNameGlob);
+
+            // null/absent for MaxPerSecond indicates unlimited, which is a negative value in the limiter
+            _limiter = MaxPerSecond is null ? new SpanRateLimiter(-1) : new SpanRateLimiter((int?)MaxPerSecond);
+        }
+
+        /// <inheritdoc/>
+        public float SamplingRate { get; } = 1.0f;
+
+        /// <inheritdoc/>
+        public float? MaxPerSecond { get; } = null;
+
+        /// <summary>
+        ///     Creates <see cref="SpanSamplingRule"/>s from the supplied JSON <paramref name="configuration"/>.
+        /// </summary>
+        /// <param name="configuration">The JSON-serialized configuration.</param>
+        /// <returns><see cref="IEnumerable{T}"/> of <see cref="SpanSamplingRule"/>.</returns>
+        public static IEnumerable<SpanSamplingRule> BuildFromConfigurationString(string configuration)
+        {
+            if (string.IsNullOrWhiteSpace(configuration))
+            {
+                return Enumerable.Empty<SpanSamplingRule>();
+            }
+
+            try
+            {
+                var rules = JsonConvert.DeserializeObject<List<SpanSamplingRuleConfig>>(configuration);
+                return rules?.Select(
+                                  rule => new SpanSamplingRule(
+                                      rule.ServiceNameGlob,
+                                      rule.OperationNameGlob,
+                                      rule.SampleRate,
+                                      rule.MaxPerSecond))
+                             .ToList() ?? Enumerable.Empty<SpanSamplingRule>();
+            }
+            catch (Exception e)
+            {
+                Log.Error(e, "Unable to parse the span sampling rule.");
+                return Enumerable.Empty<SpanSamplingRule>();
+            }
+        }
+
+        /// <inheritdoc/>
+        public bool ShouldKeep(Span span)
+        {
+            // TODO should we log here?
+            if (!IsMatch(span))
+            {
+                return false;
+            }
+
+            var sampleKeep = SamplingHelpers.SampleByRate(span.SpanId, SamplingRate);
+
+            if (!sampleKeep)
+            {
+                return false;
+            }
+
+            var limitKeep = _limiter.Allowed(span);
+
+            return limitKeep;
+        }
+
+        internal bool IsMatch(Span span)
+        {
+            if (span is null)
+            {
+                return false;
+            }
+
+            return _serviceNameRegex.Match(span.ServiceName).Success && _operationNameRegex.Match(span.OperationName).Success;
+        }
+
+        private static Regex ConvertGlobToRegex(string glob)
+        {
+            // TODO default glob (maybe null/empty/whitespace) should be *
+            var regexPattern = "^" + Regex.Escape(glob).Replace("\\?", ".").Replace("\\*", ".*") + "$";
```

Comment:
Are these the only supported globbing rules? `?` and `*`? There's no character classes or similar?